### ISSUE TITLE
Distribute password rotations more smoothly

### ISSUE THREAD

**Issue by harti2006** [zalando-stups/mint-worker#22]
The hard deadline of 120 min for password expiration, sometimes leads to all passwords being rotated at the same time (especially when mint-worker was down for a while and all passwords expired). Afterwards they are valid for the next two hours and the worker is basically idle.

If many passwords expire at the same time, the mint-worker should not try to rotate them all at once, but distribute them more smoothly, to have an almost constant workload all the time and avoid peaks.

**Comment by hjacobs** [comment]
:+1: maybe some randomization could help

**Comment by harti2006** [comment]
@hjacobs yes, or limiting the number of passwords, that are rotated with one job run.
